cmd/run.go

```diff
@@ -331,16 +319,24 @@ a commandline interface for interacting with it.`,
 					logger.Debug("Ctrl+C received, exiting...")
 				}
 			}
+
 			globalCancel() // signal the Engine that it should wind down
 			logger.Debug("Waiting for engine processes to finish...")
 			engineWait()
 			logger.Debug("Everything has finished, exiting k6!")
-			if interrupt != nil {
-				return errext.WithExitCodeIfNone(interrupt, exitcodes.ScriptException)
+
+			if runErr != nil {
+				if common.IsInterruptError(runErr) {
+					return errext.WithExitCodeIfNone(runErr, exitcodes.ScriptException)
+				}
+				return errext.WithExitCodeIfNone(runErr, exitcodes.GenericEngine)
```

Comment:
this also changes that we will show the summary on *non* InterruptError which probably is not what we want

Reply:
No? I did that intentionally as I don't see why we shouldn't show the summary in all cases.

Reply:
I see some merit to showing the summary in more cases but I think this is kind of a bigger and different change so I would prefer to have it outside of this PR 

cc @na--

Reply:
I removed this in the latest force push, so now only interrupt errors should show the summary.